backend/src/main/java/ai/verta/modeldb/datasetVersion/DatasetVersionDAORdbImpl.java

```diff
@@ -177,15 +184,28 @@ public DatasetVersionDTO getDatasetVersions(
             .setAscending(isAscending)
             .setSortKey(sortKey)
             .build();
-    return findDatasetVersions(findDatasetVersions, currentLoginUser);
+    return findDatasetVersions(datasetDAO, findDatasetVersions, currentLoginUser);
   }
 
   @Override
-  public Boolean deleteDatasetVersions(List<String> datasetVersionIds) {
+  public Boolean deleteDatasetVersions(List<String> datasetVersionIds)
+      throws InvalidProtocolBufferException {
+    List<String> accessibleDatasetVersionIds =
+        getAccessibleDatasetVersionIDs(datasetVersionIds, ModelDBServiceActions.UPDATE);
```

Comment:
```suggestion
        getAccessibleDatasetVersionIDs(datasetVersionIds, ModelDBServiceActions.DELETE);
```

Reply:
But we are also used UPDATE for the experiment run (https://github.com/VertaAI/modeldb/blob/aj/find-dataset-version-by-workspace/backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java#L382) and for the project we have used DELETE. so what is the right action?

Reply:
Experiment run has a bug then.

Reply:
initially we did not have the owner role for secondary entity. we have had this PR https://github.com/VertaAI/modeldb/pull/447/files open for a while. I would recommend updating the code here as per conrado's suggestion.  we should update #447 based on my comment from a month back there.

Reply:
Resolved